lib/rummager/app.rb

```diff
@@ -26,6 +26,12 @@ def current_index
     halt(404)
   end
 
+  def prevent_access_to_govuk
+    if index_name == 'govuk'
+      halt(403, "Actions to govuk index must be received by Rummager via the message queue, rather than the Rummager API")
+    end
+  end
+
```

Comment:
Did you look at a before filter as an option:

```
before '/*' do
  if index_name
    if index_name =~ /^mainstream/
      halt(500, "The mainstream index has been removed and further updates are blocked")
    elsif index_name =~ /^govuk/
      halt(403, "Actions to govuk index are not allowed via this endpoint, please use the message queue to update this index")
    end
  end
end
```

Reply:
@dwhenry I like the idea and think it is nice to also have a message that describes why actions can't be made to `mainstream` compared to just having the `SearchIndices::NoSuchIndex` error. I've tried to get this to work but it won't due to the params not being available to get the `index_name` from. I had a look at `before_filter` but it doesn't look like it will work for the Sinatra style routes. I can get it to work if I make it an `after` action as opposed to `before`. It does work but isn't ideal to run code before and then prevent it, but I do think it is better than what I had before. What do you think?

Reply:
We investigated this further and it is working on [this branch](https://github.com/alphagov/rummager/compare/namespace-to-prevent-govuk-access-through-rummager-api). 

The workaround was to make use of a Sinatra namespace. Unfortunately this clashed with the rack namespace. It can be made to work (see `sinatra-namespace.rb` [here](https://github.com/alphagov/rummager/commit/6cca57b9113f4c4c80f06cb0c54ece19eeb11c3c#diff-31da4034a6a3ec4e21b0ffc240a45cdb)) but we are not sure if having to understand this extra complexity for people developing on this in the future is really worth it. Therefore we think it could be better to stick with the original approach.

@MatMoore @suzannehamilton would be great to hear either of your opinions on this. Thanks!

Reply:
Your reasoning makes sense to me and I'm fine with a bit of duplication if it's simpler. These controllers don't change very often.

I can't really comment on the idiomatic-ness though as I haven't used sinatra much.

Reply:
👍